### ISSUE TITLE
full code in simulation 

### ISSUE THREAD

**Issue by AhmedTarekYA** [SallaApp/ZATCA#45]
does any one finish the full integration code ?

**Comment by thaifanisalla** [comment]
Yes , You can use this in simulation or production , Even in sandbox
we support the three modes

**Comment by AhmedTarekYA** [comment]
Unfortunately i get
+"errorMessages": array:1 [ 0 => {#3201 +"type": "ERROR" +"code":
"signed-properties-hashing" +"category": "CERTIFICATE_ERRORS" +"message":
"Invalid signed properties hashing, SignedProperties with
id='xadesSignedProperties'" +"status": "ERROR"       }     ]


في أربعاء، 6 نوفمبر، 2024 في 11:31 ص، كتب Zaher Thaifani <
***@***.***>:

**Comment by thaifanisalla** [comment]
Hi @AhmedTarekYA 
please provide us what the steps which you did?

 here some note :
 - When generate CSR, make sure to set the environment as simulation by passing 'simulation' by using 
 ````   setCurrentZatcaEnv(string $currentEnv)  function ````
- Make sure to provide the correct private key when sign invoice.
 - And for phpseclib, ensure you're using version "phpseclib/phpseclib": "3.0.x-dev".

**Comment by islamazouz** [comment]
h have the same problem 
![image](https://github.com/user-attachments/assets/f377c4cf-1d10-496d-9cfc-f5c75b16b7df)

**Comment by hamza-id** [comment]
Any resolution i am also facing the same error.

**Comment by hamza-id** [comment]
The issue relates to signed properties, specifically that their hash value does not match its calculated value. This discrepancy happens due to the formatting of signed properties before hashing. 
The signed properties hash is highly sensitive and can change with even a single whitespace modification, causing it to fail reconciliation on ZATCA side in B2C or simplified case.

**Comment by T-Mohamed-adam** [comment]
Has anyone solved this issue? I'm facing the same error: Invalid signed properties hashing, SignedProperties with id='xadesSignedProperties'. Any help would be appreciated!

**Comment by hamza-id** [comment]
Finally, I resolved this error by having 36 spaces in the first line and 4 spaces after that. This changes the hash ofthe  signed property hash value does not match its calculated value. The signed properties hash is highly sensitive and can change with even a whitespace.

```
                                     <xades:SignedSignatureProperties>
                                        <xades:SigningTime>2025-01-19T08:01:20</xades:SigningTime>
                                        <xades:SigningCertificate>
                                            <xades:Cert>
                                                <xades:CertDigest>
                                                    <ds:DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256" />
                                                    <ds:DigestValue>NTlhYjZjMDBiZmU5OGYzNmVlMmNhNzVlODhiOWQ0Nzc2ODk0ZGE4MDBkMjQ3YzgzZmQ4NDIzM2U3YzE0NWIxYQ==</ds:DigestValue>
                                                </xades:CertDigest>
                                                <xades:IssuerSerial>
                                                    <ds:X509IssuerName>0.9.2342.19200300.100.1.25=local/0.9.2342.19200300.100.1.25=gov/0.9.2342.19200300.100.1.25=extgazt, CN=PEZEINVOICESCA1-CA</ds:X509IssuerName>
                                                    <ds:X509SerialNumber>2207773857395212338642076460671910039858986559</ds:X509SerialNumber>
                                                </xades:IssuerSerial>
                                            </xades:Cert>
                                        </xades:SigningCertificate>
                                    </xades:SignedSignatureProperties>
                                </xades:SignedProperties>
```

**Comment by Prathamesh017** [comment]
can 

> Finally, I resolved this error by having 36 spaces in the first line and 4 spaces after that. This changes the hash ofthe signed property hash value does not match its calculated value. The signed properties hash is highly sensitive and can change with even a whitespace.
> 
> ```
>                                      <xades:SignedSignatureProperties>
>                                         <xades:SigningTime>2025-01-19T08:01:20</xades:SigningTime>
>                                         <xades:SigningCertificate>
>                                             <xades:Cert>
>                                                 <xades:CertDigest>
>                                                     <ds:DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256" />
>                                                     <ds:DigestValue>NTlhYjZjMDBiZmU5OGYzNmVlMmNhNzVlODhiOWQ0Nzc2ODk0ZGE4MDBkMjQ3YzgzZmQ4NDIzM2U3YzE0NWIxYQ==</ds:DigestValue>
>                                                 </xades:CertDigest>
>                                                 <xades:IssuerSerial>
>                                                     <ds:X509IssuerName>0.9.2342.19200300.100.1.25=local/0.9.2342.19200300.100.1.25=gov/0.9.2342.19200300.100.1.25=extgazt, CN=PEZEINVOICESCA1-CA</ds:X509IssuerName>
>                                                     <ds:X509SerialNumber>2207773857395212338642076460671910039858986559</ds:X509SerialNumber>
>                                                 </xades:IssuerSerial>
>                                             </xades:Cert>
>                                         </xades:SigningCertificate>
>                                     </xades:SignedSignatureProperties>
>                                 </xades:SignedProperties>
> ```

can you tell me where did you make the changes , according to the zatca-xml library

**Comment by thaifanisalla** [comment]
> Finally, I resolved this error by having 36 spaces in the first line and 4 spaces after that. This changes the hash ofthe signed property hash value does not match its calculated value. The signed properties hash is highly sensitive and can change with even a whitespace.
> 
> ```
>                                      <xades:SignedSignatureProperties>
>                                         <xades:SigningTime>2025-01-19T08:01:20</xades:SigningTime>
>                                         <xades:SigningCertificate>
>                                             <xades:Cert>
>                                                 <xades:CertDigest>
>                                                     <ds:DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256" />
>                                                     <ds:DigestValue>NTlhYjZjMDBiZmU5OGYzNmVlMmNhNzVlODhiOWQ0Nzc2ODk0ZGE4MDBkMjQ3YzgzZmQ4NDIzM2U3YzE0NWIxYQ==</ds:DigestValue>
>                                                 </xades:CertDigest>
>                                                 <xades:IssuerSerial>
>                                                     <ds:X509IssuerName>0.9.2342.19200300.100.1.25=local/0.9.2342.19200300.100.1.25=gov/0.9.2342.19200300.100.1.25=extgazt, CN=PEZEINVOICESCA1-CA</ds:X509IssuerName>
>                                                     <ds:X509SerialNumber>2207773857395212338642076460671910039858986559</ds:X509SerialNumber>
>                                                 </xades:IssuerSerial>
>                                             </xades:Cert>
>                                         </xades:SigningCertificate>
>                                     </xades:SignedSignatureProperties>
>                                 </xades:SignedProperties>
> ```

I faced these before , same solution was that , make the spaces from beginning is 
so we add some alerts [here] (https://github.com/SallaApp/ZATCA/blob/0f125ab06f31b4e78c43c312ce0d9f625f9ad49c/src/Models/InvoiceSign.php#L98) 
AND [here](https://github.com/SallaApp/ZATCA/blob/0f125ab06f31b4e78c43c312ce0d9f625f9ad49c/src/Helpers/UblExtension.php#L158)
 
<img width="657" height="433" alt="Image" src="https://github.com/user-attachments/assets/4b452cd7-2bb6-4345-b51b-2c4cbd5f63bf" />

<img width="822" height="598" alt="Image" src="https://github.com/user-attachments/assets/cb3c5590-3950-45a9-bf1f-6b4f87c5b5a9" />

**Comment by ashoeb2197** [comment]
I uses the package and it works fine in production mode but in simulation it reports the same issue is that normal??
